tests/kernel/interrupt/src/interrupt_common.c

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <kernel_structs.h>
+#include "interrupt_util.h"
+
+#if defined(CONFIG_X86)
+
+#define IRQ0_LINE       16
+#define IRQ1_LINE       17
+#define ISR0_ARG        0xca55e77e
+#define ISR1_ARG	0xb01dface
+
+#define TEST_VECTOR_0	(IRQ0_LINE + 32)
+#define TEST_VECTOR_1	(IRQ1_LINE + 32)
+
+static ulong_t handler_test_result;
+volatile ulong_t int_0_esp;
+volatile ulong_t int_1_esp;
+
+/* this helper function gets esp of current thread */
+static void get_stack_addr(volatile ulong_t *reg)
+{
+	__asm__ volatile("mov %%esp,%0" : "=m"(*reg) : );
+}
+
+/* this helper function trigger self-defined interrupt */
+static void trigger_interrupt(int int_case)
+{
+	switch (int_case) {
+	case IRQ0_LINE:
+		__asm__ volatile("int %0" : : "i" (TEST_VECTOR_0));
+		break;
+	case IRQ1_LINE:
+		__asm__ volatile("int %0" : : "i" (TEST_VECTOR_1));
+		break;
+	default:
+		break;
+	}
+}
+
+/* self-defined handler for testing */
+static void test_handler(void *param)
+{
+	handler_test_result = (ulong_t)param;
+
+	/* for different ISR, get esp of current thread */
+	if (handler_test_result == ISR0_ARG) {
+		get_stack_addr(&int_0_esp);
+	}
+
+	if (handler_test_result == ISR1_ARG) {
+		get_stack_addr(&int_1_esp);
+	}
+}
+
+/**
+ * @brief Test interrupt features
+ *
+ * @details This test case validates four features of interrupt
```

Comment:
This seems to me cross-ARCH; can this be done for the other architectures, or is it done already?

Reply:
Most parts of this case can be found in other test cases, such as configuration of interrupts at build time/runtime, or pass argument to interrupt handler. I saw them in tests/kernel/interrupt, tests/kernel/gen_isr_table, etc.
 Should I implement this test case to cross-ARCH? Any suggestion for this?

Reply:
This file name has been updated. Fixed.